pyanaconda/modules/storage/partitioning/blivet_interface.py

```diff
@@ -17,11 +17,65 @@
 # License and may only be used or replicated with the express permission of
 # Red Hat, Inc.
 #
+import pickle
+
+from blivetgui.communication.server import BlivetUtilsServer
+
 from pyanaconda.dbus.interface import dbus_interface
+from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
 from pyanaconda.modules.common.constants.objects import BLIVET_PARTITIONING
 from pyanaconda.modules.storage.partitioning.base_interface import PartitioningInterface
 
 
 @dbus_interface(BLIVET_PARTITIONING.interface_name)
 class BlivetPartitioningInterface(PartitioningInterface):
     """DBus interface for the Blivet partitioning module."""
+
+    def SendRequest(self, data: List[Byte]) -> List[Byte]:
+        """Send a request to the storage handler.
+
+        :param data: a request data in bytes
+        :return: a reply data in bytes
+        """
+        server = AnacondaUtilsServer(self.implementation.handler, bytes(data))
+        return server.handle()
+
+
+class AnacondaUtilsServer(BlivetUtilsServer):
+    """The request handler for the Blivet."""
+
+    def __init__(self, handler, data):  # pylint: disable=super-init-not-called
+        self.blivet_utils = handler
+        self._data = data
+        self._result = None
+
+    def handle(self):
+        """Handle a message."""
+        msg = self._recv_msg()
+
+        unpickled_msg = pickle.loads(msg)
+
+        if unpickled_msg[0] == "call":
+            self._call_utils_method(unpickled_msg)
+
+        elif unpickled_msg[0] == "param":
+            self._get_param(unpickled_msg)
+
+        elif unpickled_msg[0] == "method":
+            self._call_method(unpickled_msg)
+
+        elif unpickled_msg[0] == "next":
+            self._get_next(unpickled_msg)
+
```

Comment:
Please remove the spaces between `if-else`.

Reply:
Fixed.